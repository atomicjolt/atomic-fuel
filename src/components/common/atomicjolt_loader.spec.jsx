import React from 'react';
import { render } from '@testing-library/react';
import Loader, { CoreLoader } from './atomicjolt_loader';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

const mockStore = configureStore([]);
const store = mockStore({});

describe('Loader', () => {
  it('renders the default loader', () => {
    const { container } = render(
      <Provider store={store}>
        <Loader message="Loading..." />
      </Provider>,
    );

    expect(container).toMatchSnapshot();
  });

  it('renders the core loader', () => {
    const { container } = render(
      <Provider store={store}>
        <CoreLoader message="Loading..." />
      </Provider>,
    );

    expect(container).toMatchSnapshot();
  });
});
