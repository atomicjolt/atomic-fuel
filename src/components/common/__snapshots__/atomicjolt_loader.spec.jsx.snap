// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader renders the core loader 1`] = `
<div>
  <div
    class="aj-loader"
  >
    <div
      class="atomicjolt-loading-animation"
    >
      <svg
        aria-label="loading"
        role="img"
        viewBox="0 0 91.87 114.09"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          data-name="Layer 2"
        >
          <polygon
            class="cls-1"
            points="40.45 111.32 89.11 99.26 71.35 19.9 21.1 89.71 40.45 111.32"
          />
          <polyline
            class="cls-2"
            points="50.67 2.77 2.77 69.96 25.47 94.65 66.36 84.13 50.67 2.77 71.35 19.9"
          />
        </g>
      </svg>
    </div>
    <p
      class="loader-text"
    >
      Loading...
    </p>
  </div>
</div>
`;

exports[`Loader renders the default loader 1`] = `
<div>
  <div
    class="aj-loader"
  >
    <div
      class="atomicjolt-loading-animation"
    >
      <svg
        aria-label="loading"
        role="img"
        viewBox="0 0 91.87 114.09"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          data-name="Layer 2"
        >
          <polygon
            class="cls-1"
            points="40.45 111.32 89.11 99.26 71.35 19.9 21.1 89.71 40.45 111.32"
          />
          <polyline
            class="cls-2"
            points="50.67 2.77 2.77 69.96 25.47 94.65 66.36 84.13 50.67 2.77 71.35 19.9"
          />
        </g>
      </svg>
    </div>
    <p
      class="loader-text"
    >
      Loading...
    </p>
  </div>
</div>
`;
